### PR TITLE
Efficiency as a programmer

### PR DESCRIPTION
**Pull request by smonn**
This is mostly a brain dump of things I could think of that has helped me over my 10+ years as a programmer. Certainly not an exhaustive list of things, but hopefully enough to get you into the mindset. Let me know what you think!

### CODE REVIEW

File: soapbox/general/smonn-efficiency.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+---
+
+Someone once said that a great programmer is a lazy one. The key being that they try to get as much done with as little effort as possible as opposed to not getting anything done. In other words, they are efficient. So what can you do to improve your efficiency?
+
+## TL;DR
+
+- Learn and use keyboard shortcuts
+- Write less/more code
+- Ask questions early and often
```
Comment (theothermattm) [inline soapbox/general/smonn-efficiency.md:15]: ❤️

Comment (theothermattm) [reply soapbox/general/smonn-efficiency.md:15]: A topic near and dear to my heart; https://code.theothermattm.com/ask-why/

File: soapbox/general/smonn-efficiency.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+- Sometimes you just gotta RTFM
+- Always be learning
+
+## Learn and use keyboard shortcuts
+
+This applies to any editor (or IDE). As soon as you find yourself doing something repetitive that _feels_ slow, try to find out if there's a keyboard shortcut for it. If not, can you add one?
+
```
Comment (theothermattm) [inline soapbox/general/smonn-efficiency.md:22]: I feel like it's worth a mention to invest time in learning a keyboard based editor such as Vi or Emacs here.

File: soapbox/general/smonn-efficiency.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+For example, naming is hard. So eventually you'll want to rename a [symbol][symbol]. If your choice of editor includes some basic refactoring tools, chances are you can easily trigger a rename refactoring. In VS Code, _the best editor_, you can simply hit F2 with the symbol you want to rename selected. This should then update all references to that symbol to use the new name.
+
+[symbol]: https://en.wikipedia.org/wiki/Symbol_(programming)
+
+## Write less/more code
```
Comment (theothermattm) [inline soapbox/general/smonn-efficiency.md:27]: What about mentioning the art of _deleting_ code??

**Comment by theothermattm** [comment]
I love this so much.

One topic i would personally add (so take or leave the advice, obviously, I won't be offended) would be "Know what _not_ to build." This goes hand in hand with asking questions early and often.  Sometimes if you ask the riht questions something can be totally omitted saving time for you to focus on the things that really matter.

**Review by jakerainis — APPROVED** [review]
This looks great Simon. I really enjoyed reading and need to remind myself to `rtfm` more!

File: soapbox/general/smonn-efficiency.md (hunk 2 above)
Comment (smonn) [reply soapbox/general/smonn-efficiency.md:22]: Probably for over-achievers, but it is useful to know some basic terminal-based editor like Vi[m] (or even Nano). Never know when you end up on a server via SSH and need to edit that one config file... So will add a mention of this!

File: soapbox/general/smonn-efficiency.md (hunk 3 above)
Comment (smonn) [reply soapbox/general/smonn-efficiency.md:27]: Great point! Will see if I can bake it into this section or make it its own.

**Comment by smonn** [comment]
> One topic i would personally add (so take or leave the advice, obviously, I won't be offended) would be "Know what not to build." This goes hand in hand with asking questions early and often. Sometimes if you ask the riht questions something can be totally omitted saving time for you to focus on the things that really matter.

Totally! Was sort of one of the things I had in mind, but didn't quite put it like that. Will reword or add a paragraph to emphasize this better.

Thanks @theothermattm and @jakerainis!

**Comment by theothermattm** [comment]
@smonn could we resubmit this to https://github.com/deptagency/engineering-soapboxes when you make your updates?

**Comment by theothermattm** [comment]
Thanks for moving this over! :)

**Pull request closed without merging**